crates/rattler_auth/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+use std::str::FromStr;
+
+use keyring::{Entry, Result};
+use reqwest::{Request, Client, IntoUrl};
+
+#[derive(Debug)]
+pub enum Authentication {
+    BearerToken(String),
+    Basic{username: String, password: String},
+    CondaToken(String),
+}
+
+#[derive(Debug)]
+pub enum AuthenticationParseError {
+    InvalidScheme,
+    InvalidToken,
+}
+
+impl FromStr for Authentication {
+    type Err = AuthenticationParseError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let mut parts = s.split_whitespace();
+        let scheme = parts.next().unwrap();
+        let token = parts.next().unwrap();
+        match scheme {
+            "Bearer" => Ok(Authentication::BearerToken(token.to_string())),
+            "Basic" => {
+                let mut token_parts = token.split(":");
+                let username = token_parts.next().unwrap();
+                let password = token_parts.next().unwrap();
+                Ok(Authentication::Basic{username: username.to_string(), password: password.to_string()})
+            },
+            "CondaToken" => Ok(Authentication::CondaToken(token.to_string())),
+            _ => Err(AuthenticationParseError::InvalidScheme),
+        }
+    }
+}
+
+pub fn store_authentication_entry(host: &str, authentication: &Authentication) -> Result<()> {
+    let entry = Entry::new("rattler_auth", host)?;
+    match authentication {
+        Authentication::BearerToken(token) => {
+            let password = format!("Bearer {}", token);
```

Comment:
Can you implement Display? Since you also implement FromStr it makes sense to be able to convert from String and back.